### ISSUE TITLE
Parser: stray closer keywords (`done`/`fi`/`esac`) silently accepted

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#1362]
### Summary
A bare loop or branch closer keyword (`done`, `fi`, `esac`) used as a statement with no matching opener is silently accepted; Zsh rejects it. This is a missing-diagnostic (false-negative) gap.

### Reproduction
```zsh
echo hi; done
```
`zsh -n` reports `parse error near \`done'`. `zshellcheck` reports zero parser errors. The same holds for bare `done`, `fi`, `esac`.

### Expected
A reserved closer with no opener is a syntax error (`man zshmisc`, RESERVED WORDS). The linter should surface it rather than pass it silently.

### Environment
- zshellcheck v1.1.0

**Comment by afadesigns** [comment]
Attempted a statement-head guard (reject `done`/`fi`/`esac` when dispatched as a statement) and reverted it. The closer-as-terminator entries in `expressionTerminators` are load-bearing for block nesting and command-argument handling: a naive guard breaks valid input such as `echo done` (where `done` is a literal argument the terminator set bounces), nested `if ... fi; fi`, and pipeline tails like `... done || echo done`. A correct fix needs command-position vs argument-position awareness so a closer keyword is rejected only at a true statement head with no opener, never as an ordinary word. Staying deferred.

**Comment by afadesigns** [comment]
Resolved in v1.3.3. An orphan compound closer (`fi`, `done`, `esac`) at the top level of a script is now reported as a parse error instead of being silently accepted. The check runs only at program top level; a closer nested inside a body can be a parse-desync artefact, so flagging it there would risk a false positive. Regression tests cover the top-level strays and the valid nested forms.
